core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -229,6 +270,30 @@ protected final void requestMessagesFromDeframer(int numMessages) {
     }
   }
 
+  /**
+   * Event handler to be called by the subclass when a number of bytes are being queued for sending
+   * to the remote endpoint.
+   *
+   * @param numBytes the number of bytes being sent.
+   */
+  protected final void onSendingBytes(int numBytes) {
+    numSentBytesQueued += numBytes;
```

Comment:
We should probably call `onReady()` here if `numSentBytesQueued < onReadyThreshold` ? We don't want to wait until the write actually completes.